src/OrchardCore.Modules/OrchardCore.GitHub/Configuration/GitHubAuthenticationOptionsConfiguration.cs

```diff
@@ -1,57 +1,33 @@
 using System.Diagnostics;
-using Microsoft.AspNetCore.Authentication;
+using AspNet.Security.OAuth.GitHub;
 using Microsoft.AspNetCore.DataProtection;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using OrchardCore.GitHub.Settings;
 
 namespace OrchardCore.GitHub.Configuration;
 
-public class GitHubOptionsConfiguration :
-    IConfigureOptions<AuthenticationOptions>,
-    IConfigureNamedOptions<GitHubOptions>
+public class GitHubAuthenticationOptionsConfiguration :
+    IConfigureNamedOptions<GitHubAuthenticationOptions>
 {
     private readonly GitHubAuthenticationSettings _gitHubAuthenticationSettings;
     private readonly IDataProtectionProvider _dataProtectionProvider;
     private readonly ILogger _logger;
 
-    public GitHubOptionsConfiguration(
+    public GitHubAuthenticationOptionsConfiguration(
         IOptions<GitHubAuthenticationSettings> gitHubAuthenticationSettings,
         IDataProtectionProvider dataProtectionProvider,
-        ILogger<GitHubOptionsConfiguration> logger)
+        ILogger<GitHubAuthenticationOptionsConfiguration> logger)
     {
         _gitHubAuthenticationSettings = gitHubAuthenticationSettings.Value;
         _dataProtectionProvider = dataProtectionProvider;
         _logger = logger;
     }
 
-    public void Configure(AuthenticationOptions options)
-    {
-        if (_gitHubAuthenticationSettings == null)
-        {
-            return;
-        }
-
-        if (string.IsNullOrWhiteSpace(_gitHubAuthenticationSettings.ClientID) ||
```

Comment:
@kevinchalet  With the simpler registration, we lose this logic.

Reply:
Yeah, we definitely don't want that for the reasons I mentioned in my other message 😄